bfvmm/src/vmcs/src/vmcs_intel_x64.cpp

```diff
@@ -496,39 +543,39 @@ vmcs_intel_x64::secondary_processor_based_vm_execution_controls()
 void
 vmcs_intel_x64::vm_exit_controls()
 {
-    auto controls = vm::read(VMCS_VM_EXIT_CONTROLS);
-
-    controls |= VM_EXIT_CONTROL_SAVE_DEBUG_CONTROLS;
-    controls |= VM_EXIT_CONTROL_HOST_ADDRESS_SPACE_SIZE;
-    controls |= VM_EXIT_CONTROL_LOAD_IA32_PERF_GLOBAL_CTRL;
-    controls |= VM_EXIT_CONTROL_ACKNOWLEDGE_INTERRUPT_ON_EXIT;
-    controls |= VM_EXIT_CONTROL_SAVE_IA32_PAT;
-    controls |= VM_EXIT_CONTROL_LOAD_IA32_PAT;
-    controls |= VM_EXIT_CONTROL_SAVE_IA32_EFER;
-    controls |= VM_EXIT_CONTROL_LOAD_IA32_EFER;
-    // controls |= VM_EXIT_CONTROL_SAVE_VMX_PREEMPTION_TIMER_VALUE;
+    auto controls = vmcs::vm_exit_controls::get();
+
+    controls |= vmcs::vm_exit_controls::save_debug_controls::mask;
```

Comment:
These all need to be converted. The approach here is just wrong (both the old one and this one). Instead, each control should be set using the `enable` function with the exception to this being the could in the secondary controls like XSAVE, RDTSCP and the invalidation instruction which should be `enable_if_allowed`.

Just like I had before, enable all of them using the same scheme, and then comment out the ones we don't use.